File: Check If All 1's Are at Least Length K Places Away/solution.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        currOne = -1
        for i, num in enumerate(nums):
            if num == 1:
                currOne = i
                break
        if currOne == -1: return True
        for i in range(currOne + 1, len(nums)):
            if nums[i] == 1:
                if i - currOne - 1 < k:
                    return False
                currOne = i
        return True

File: Check If All 1's Are at Least Length K Places Away/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var kLengthApart = function (nums, k) {
    let currOne = -1;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 1) {
            currOne = i;
            break;
        }
    }
    if (currOne === -1) return true;
    for (let i = currOne + 1; i < nums.length; i++) {
        if (nums[i] === 1) {
            if (i - currOne - 1 < k) return false;
            currOne = i;
        }
    }
    return true;
};